pandas/tests/reshape/test_concat.py

```diff
@@ -1294,6 +1295,17 @@ def test_concat_rename_index(self):
         tm.assert_frame_equal(result, exp)
         assert result.index.names == exp.index.names
 
+    def test_concat_with_ordered_dict(self):
+        # GH 21510
+        result = pd.concat(OrderedDict([('First', pd.Series(range(3))),
+                                        ('Another', pd.Series(range(4)))]))
+        a = pd.Series(range(3), range(3))
+        b = pd.Series(range(4), range(4))
+        a.index = pd.MultiIndex.from_tuples([('First', v) for v in a.index])
+        b.index = pd.MultiIndex.from_tuples([('Another', v) for v in b.index])
+        expected = a.append(b)
```

Comment:
It would be easier if you just constructed the Series directly. You could doing something like `index = pd.MultiIndex.from_records(...)` and then `expected = pd.Series(..., index=index)` to make this as succinct and clear as possible

Reply:
AttributeError: type object 'MultiIndex' has no attribute 'from_records'